### ISSUE TITLE
Work-around if using react-snap

### ISSUE THREAD

**Issue by rscotten** [t49tran/react-google-recaptcha-v3#12]
Hello,

Thanks for making this library.  

Unfortunately, I found that it doesn't work with the `react-snap` library and `create-react-app`.  I'm using the `withGoogleReCaptcha` HOC, and the `executeRecaptcha()` function doesn't execute.  It also throws no error, so it's not easy to debug.

I wanted to post my workaround in case anybody runs into this issue.

The given code example in the Readme goes like this:

```javascript
import { GoogleReCaptchaProvider } from 'react-google-recaptcha-v3';
 
ReactDom.render(
  <GoogleReCaptchaProvider
    reCaptchaKey="[Your recaptcha key]"
    language="[optional_language]"
  >
    <YourApp />
  </GoogleReCaptchaProvider>,
  document.getElementById('app')
);
```

My workaround is this:

```typescript
const MyReCaptchaProvider = ({ children }: any) => {
  if (navigator.userAgent === 'ReactSnap') return children;

  return (
    <GoogleReCaptchaProvider
      reCaptchaKey="[Your recaptcha key]"
      language="[optional_language]"
    >
      {children}
    </GoogleReCaptchaProvider>
  );
};

ReactDom.render(
  <MyReCaptchaProvider>
    <YourApp />
  </MyReCaptchaProvider>,
  document.getElementById('app')
);
```

Basically, the work-around tells React-Snap to ignore the GoogleReCaptchaProvider when pre-rendering the page at build time.  But when the website loads in the browser and is hydrated, then the GoogleReCaptchaProvider is initialized.  

Hope this helps someone.

**Comment by t49tran** [comment]
Hi @rscotten, thanks for that. I haven't looked into how this lib work with SSR solutions. What you have posted seems really useful and at least we can consider to add it to the docs. I completely believe that the provider should be able to work properly in these situations.

Let me see how the libs work with couple of SSR setups before decide what we needs to do. In the meantime, let me know if you would like to add this section to the README.

**Comment by Nash0x7E2** [comment]
Does this workaround also work for create-react-app? The banner is shown on the side of the screen however the widget is not rendered. 

@salvoravida Looked at the `userAgent` however it does not match ReactSnap at any point so the condition is never being met. 

Any advice on this issue would be greatly appreciated.

**Comment by rscotten** [comment]
@Nash0x7E2 The React Snap library is a NodeJS pre-rendering library.  My workaround is designed for using Create React App (CRA) when pre-rendering with React-Snap.  

`react-google-recaptcha-v3` should work just fine with CRA.  The userAgent will be whatever browser you're using.

**Comment by spencer741** [comment]
@rscotten I am having the same issue with CRA as @Nash0x7E2 is. 

The component is not showing up at all for me. I am also getting the **Protected by reCaptcha** banner in the bottom right hand corner. In addition, the recaptcha http request  to Google is successful. After inspecting the DOM, the component is not getting rendered at all.

**Comment by kajchang** [comment]
@Nash0x7E2 @spencer741 your problem is unrelated to this one. There's no widget because Recaptcha v3 doesn't rely on a challenge widget, you should use Recaptcha v2 if you want to use a challenge widget. This confused me a bit initially.

https://developers.google.com/recaptcha/docs/faq

**Comment by Nash0x7E2** [comment]
Thanks @kajchang, in the end I did switch to V2 and that solved my problem.
